### DISCUSSION TITLE
Update GraphRAG to support LazyGraphRAG

### DISCUSSION THREAD

**Discussion by sbmandava** [infiniflow/ragflow#3843] [Ideas]
Feature Enhancement : 
- Improve GraphRAG performance, quality and cost of tokens consumed. 
- Make LazyGraphRAG the default mode for GraphRAG.

LazyGraphRAG: Setting a new standard for quality and cost 
Microsoft Nov 15, 2024
https://www.microsoft.com/en-us/research/blog/lazygraphrag-setting-a-new-standard-for-quality-and-cost/

* LazyGraphRAG data indexing costs are identical to vector RAG and 0.1% of the costs of full GraphRAG. 
* For comparable query costs to vector RAG, LazyGraphRAG outperforms all competing methods on local queries, including long-context vector RAG and GraphRAG [DRIFT](https://www.microsoft.com/en-us/research/blog/introducing-drift-search-combining-global-and-local-search-methods-to-improve-quality-and-efficiency/) search (our recently introduced RAG approach shown to outperform vector RAG) as well as GraphRAG local search. 
* The same LazyGraphRAG configuration also shows comparable answer quality to GraphRAG Global Search for global queries, but more than 700 times lower query cost.
* For 4% of the query cost of GraphRAG global search, LazyGraphRAG significantly outperforms all competing methods on both local and global query types, including GraphRAG global search at the C2 level (the third level of the community hierarchy recommended for most applications).

**Comment by KevinHuSh** [comment]
#3862
